beetsplug/edit.py

```diff
@@ -0,0 +1,438 @@
+
+    def get_selected_fields(self, myfields, objs, opts):
+        a = []
+        if opts.sum:
+            for field in myfields:
+                if field not in self.not_fields:
+                    d = collections.defaultdict(str)
+                    for obj in objs:
+                        d[obj[field]] += (" " + str(obj['id']))
+                    a.append([field, [{f: i} for f, i in d.items()]])
+            return a
+        else:
+            for mod in objs:
+                a.append([{fi: mod[fi]}for fi in myfields])
+            return a
+
+    def get_all_fields(self, myfields, objs, opts):
+        a = []
+        if opts.sum:
+            fields = (library.Album.all_keys() if opts.album
+                      else library.Item.all_keys())
+            for field in sorted(fields):
+                # for every field get a dict
+                d = collections.defaultdict(str)
+                for obj in objs:
+                    # put all the ob-ids in the dict[field] as a string
+                    d[obj[field]] += (" " + str(obj['id']))
+                # for the field, we get the value and the users
+                a.append([field, [{f: i} for f, i in sorted(d.items())]])
+            return a
+        else:
+            for mod in objs:
+                a.append([{fi: mod[fi]} for fi in sorted(mod._fields)])
+            return a
+
+    def change_objs(self, dict_items):
+        # construct a yaml from the original object-fields
+        # and make a yaml that we can change in the text-editor
+        oldyaml = self.print_items[self.style](dict_items)
+        newyaml = self.print_items[self.style](dict_items)
+        new = NamedTemporaryFile(suffix='.yaml', delete=False)
+        new.write(newyaml)
+        new.close()
+        self.get_editor(new.name)
+
+        if ui.input_yn(ui.colorize('action_default', "done?(y)"), True):
+            while True:
+                try:
+                    with open(new.name) as f:
+                        newyaml = f.read()
+                        list(yaml.load_all(newyaml))
+                        break
+                except yaml.YAMLError as e:
+                    print_(ui.colorize('text_warning',
+                           "change this fault: {}".format(e)))
+                    print_("correct format for empty = - '' :")
+                    if ui.input_yn(
+                            ui.colorize('action_default', "fix?(y)"), True):
+                        self.get_editor(new.name)
+                        if ui.input_yn(ui.colorize(
+                           'action_default', "ok.fixed.(y)"), True):
+                            pass
+
+            return newyaml, oldyaml
+        else:
+            exit()
+
+    def get_editor(self, name):
+        if not self.ed:
+            editor = os.getenv('EDITOR')
+            if editor:
+                os.system(editor + " " + name)
+            else:
+                webbrowser.open(name, new=2, autoraise=True)
```

Comment:
Just a quick question in regards to the safety of using `webbrowser.open` I came across while working on my previous set of changes: has it been reasonably tested that it really produces the desired results? Specially, considering the remark on [the python docs](https://docs.python.org/2/library/webbrowser.html#webbrowser.open):

>  Note that on some platforms, trying to open a filename using this function, may work and start the operating system’s associated program. However, this is neither supported nor portable.

For example, in my home machine (debian 8 + libreoffice without too much tuning), seems that debian by default does not add the `$EDITOR` variable to the environment, and as a consequence `webbrowser.open` launchs libreoffice writer.

If associating .yaml files with vim via `$ xdg-mime default vim.desktop application/x-yaml`, webbrowser seems to exhibit some weird behaviour, launching vim but not replying to stdin correctly and resulting in an unresponsive vim - after killing it manually, it launches libreoffice once again. Might be worth devoting some time to check it more carefully, as it might affect more users and the `config` command as well?

Perhaps it would be enough to add a clearer reminder to the users on the documentation to set `$EDITOR`, to avoid nuisances? It's already on http://beets.readthedocs.org/en/v1.3.15/reference/cli.html#config anyway, but might not hurt.